nibabel/gifti/tests/test_gifti.py

```diff
@@ -24,6 +26,35 @@ def test_gifti_image():
     gi = GiftiImage()
     assert_equal(gi.darrays, [])
 
+    # Test darrays / numDA
+    gi = GiftiImage()
+    assert_equal(gi.numDA, 0)
+
+    da = GiftiDataArray(data='data')
+    gi.add_gifti_data_array(da)
+    assert_equal(gi.numDA, 1)
+    assert_equal(gi.darrays[0].data, 'data')
+
+    gi.remove_gifti_data_array(0)
+    assert_equal(gi.numDA, 0)
+
+    # Remove from empty
+    gi = GiftiImage()
+    gi.remove_gifti_data_array_by_intent(0)
+    assert_equal(gi.numDA, 0)
+
+    # Remove one
+    gi = GiftiImage()
+    da = GiftiDataArray(data='data')
+    gi.add_gifti_data_array(da)
+
+    gi.remove_gifti_data_array_by_intent(0)
+    assert_equal(gi.numDA, 1)
+
+    gi.darrays[0].intent = 0
+    gi.remove_gifti_data_array_by_intent(0)
+    assert_equal(gi.numDA, 0)
+
```

Comment:
Maybe a check that trying to remove a data array when `numDA == 0` produces an `IndexError` and maintains a `numDA == 0`? At least with the old version where the number was incremented and decremented manually, I'd want to be sure it can't go negative.